[CI] AzureStorageRepositoryClientYamlTestSuiteIT( repository_azure/10_repository:56)

Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob+third-party-tests-azure-sas/189/console
https://gradle-enterprise.elastic.co/s/uxlrgod35ayr4

```
./gradlew ':plugins:repository-azure:qa:microsoft-azure-storage:integTestRunner' --tests "org.elasticsearch.repositories.azure.AzureStorageRepositoryClientYamlTestSuiteIT.test {yaml=repository_azure/10_repository/Snapshot/Restore with repository-azure}" \
  -Dtests.seed=2AD384511D4F359C \
  -Dtests.security.manager=true \
  -Dtests.locale=ko-KR \
  -Dtests.timezone=America/Scoresbysund \
  -Dcompiler.java=13 \
  -Druntime.java=8
```

Doesn't reproduce for me.

```
org.elasticsearch.repositories.azure.AzureStorageRepositoryClientYamlTestSuiteIT > test {yaml=repository_azure/10_repository/Snapshot/Restore with repository-azure} FAILED
12:03:29     java.lang.AssertionError: Failure at [repository_azure/10_repository:56]: expected [2xx] status code but api [snapshot.create] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"invalid_snapshot_name_exception","reason":"[repository:snapshot-one] Invalid snapshot name [snapshot-one], snapshot with the same name already exists","stack_trace":"InvalidSnapshotNameException[[repository:snapshot-one] Invalid snapshot name [snapshot-one], snapshot with the same name already exists]\n\tat org.elasticsearch.snapshots.SnapshotsService$2.lambda$doRun$1(SnapshotsService.java:436)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)\n\tat org.elasticsearch.common.util.concurrent.ListenableFuture$1.doRun(ListenableFuture.java:112)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:225)\n\tat org.elasticsearch.common.util.concurrent.ListenableFuture.notifyListener(ListenableFuture.java:106)\n\tat org.elasticsearch.common.util.concurrent.ListenableFuture.addListener(ListenableFuture.java:68)\n\tat org.elasticsearch.action.StepListener.whenComplete(StepListener.java:78)\n\tat org.elasticsearch.snapshots.SnapshotsService$2.doRun(SnapshotsService.java:432)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"}],"type":"invalid_snapshot_name_exception","reason":"[repository:snapshot-one] Invalid snapshot name [snapshot-one], snapshot with the same name already exists","stack_trace":"InvalidSnapshotNameException[[repository:snapshot-one] Invalid snapshot name [snapshot-one], snapshot with the same name already exists]\n\tat org.elasticsearch.snapshots.SnapshotsService$2.lambda$doRun$1(SnapshotsService.java:436)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)\n\tat org.elasticsearch.common.util.concurrent.ListenableFuture$1.doRun(ListenableFuture.java:112)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:225)\n\tat org.elasticsearch.common.util.concurrent.ListenableFuture.notifyListener(ListenableFuture.java:106)\n\tat org.elasticsearch.common.util.concurrent.ListenableFuture.addListener(ListenableFuture.java:68)\n\tat org.elasticsearch.action.StepListener.whenComplete(StepListener.java:78)\n\tat org.elasticsearch.snapshots.SnapshotsService$2.doRun(SnapshotsService.java:432)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"},"status":400}]
```

This qa module only seems to create a single snaphot, but somehow it already exists.

--- a/plugins/repository-azure/qa/microsoft-azure-storage/src/test/resources/rest-api-spec/test/repository_azure/10_repository.yml
+++ b/plugins/repository-azure/qa/microsoft-azure-storage/src/test/resources/rest-api-spec/test/repository_azure/10_repository.yml
@@ -13,6 +13,19 @@ setup:
             client: "integration_test"
             base_path: ${base_path}
 
+  # Remove the snapshots, if a previous test failed to delete them. This is
+  # useful for third party tests that runs the test against a real external service.
+  - do:
+      snapshot.delete:
+        repository: repository
+        snapshot: snapshot-one
+        ignore: 404
+  - do:
+      snapshot.delete:
+        repository: repository
+        snapshot: snapshot-two
+        ignore: 404
+
 ---
 "Snapshot/Restore with repository-azure":
   - skip:

